_posts/2019-10-21-vscode-quarkus-0.1.0.adoc

```diff
@@ -0,0 +1,141 @@
+* link:#formatting-support-for-application-properties[Formatting support for application.properties]
+* link:#retrieve-test-properties-for-an-application-properties-file-in-srctestresources[Retrieve test properties for an application.properties file in `src/test/resources`]
+* link:#new-snippets[New snippets (Quarkus datasource & Jaeger tracer)]
+
+{blank}
+
+=== Value type validation for application.properties
+For property values, there will now be an error diagnostic
+if the expected type does not match the value type.
+
+image::01_type_validation.gif[alt="Validate property value types"]
+
+For the example above, `quarkus.http.cors` expects a `boolean` and
+`quarkus.http.port` expects an `int`.
+
+The `item.price` config property was defined in a resource class to 
+expect a `float` like so: 
+
+.MyResource.java
+[source,java]
+----
+@ConfigProperty(name = "item.price")
+float price;
+----
+
+{blank}
+
+=== Go to definition support for config properties in application.properties
+In VS Code, the 'Go to definition' feature is used by holding Ctrl (or ⌘) and 
+left clicking on the desired config property, or by right-clicking the 
+desired config property and clicking ‘Go to definition’ in the context menu.
+
+This feature supports config properties defined in .jar files (`quarkus.https.cors`)
+and source files (`item.price`).
+
+image::02_go_to_definition.gif[alt="Go to definition"]
+
+{blank}
+
+=== Formatting support for application.properties
+Thanks to formatting support, it is now possible to
+quickly remove extra whitespace or lines that may
+cluttering your application.properties file.
+
+image::03_format.gif[alt="Format application.properties"]
+
+Another use case for formatting, is adding/removing spacing
+around the equals sign.
+To add/remove spacing around the equals sign, change
+the `quarkus.tools.formatting.surroundEqualsWithSpaces`
+setting to `true`/`false` in the VS Code settings.
+Then, format the document.
+
+image::04_format_spaces.gif[alt="Add spaces around equals sign"]
+
+{blank}
+
+=== Retrieve test properties for an application.properties file in `src/test/resources`
+An application.properties file located under the
+`src/test/resources` directory will now receive property
+keys coming from test scoped dependencies and source files.
+This means that completion options, documentation and
+hover will provide relevant property keys for that
+application.properties file.
+
+For example, suppose you have the following config property
+defined in a test resource class:
+
+.MyTestResource.java
+[source,java]
+----
+@ConfigProperty(name = "greeting.test", defaultValue = "testing")
+String test;
+----
+
+Also, suppose you have the following test-scoped dependency
+in your pom.xml:
+
+.pom.xml
+[source,xml]
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-hibernate-orm-panache</artifactId>
+    <scope>test</scope>
```

Comment:
I don't see the use case for having panache only in test. Do you have something in mind. I'd prefer if the use case was explicit so it can resonate better to the user.

Reply:
Thanks for the feedback,

A use case is explained here: https://github.com/redhat-developer/quarkus-ls/issues/5, specifically this part:

> As a use case, imagine an application.properties in src/test/resources, that would be able to use a specific test database (from a quarkus extension available in the test scope), in order to override a production DB defined in src/main/resources/application.properties

@fbricon does it make sense to have `quarkus-hibernate-orm-panache` only in test scope?